securedrop-admin

```diff
@@ -0,0 +1,120 @@
+"""
+
+
+import argparse
+import subprocess
+import os
+import sys
+import getpass
+
+SD_DIR = os.path.dirname(os.path.realpath(__file__))
+ANSIBLE_PATH = os.path.join(SD_DIR, "./install_files/ansible-base")
+SITE_CONFIG = os.path.join(ANSIBLE_PATH, "group_vars/all/site-specific")
+VENV_DIR = os.path.join(SD_DIR, ".venv")
+VENV_ACTIVATION = os.path.join(VENV_DIR, 'bin/activate_this.py')
+
+def error(msg, level='warning', exit=True):
+    """ Stupid wrapper function to normalize warnings """
+    print("{} - {}".format(level.upper(), msg))
+    if exit:
+        sys.exit(1)
+
+def activate_venv():
+    """Use to activate the local virtualenv"""
+    try:
+        execfile(VENV_ACTIVATION, dict(__file__=VENV_ACTIVATION))
+    except FileNotFoundError:
+        error("Pre-requisites not in place, re-run command with 'setup'",
+                level='error')
+
+def sdconfig(args):
+    """Configure SD site settings"""
+    activate_venv()
+    subprocess.call(['./configure'], cwd=ANSIBLE_PATH, env=os.environ.copy())
+
+def envsetup(args):
+    """Install SD pre-reqs"""
+    # virtualenv doesnt exist? Install dependencies and create
+    if not os.path.exists(VENV_ACTIVATION):
+        subprocess.call(['sudo', 'su', '-c', 
+                        "apt-get update && \
+                        apt-get install -y \
+                        python-virtualenv \
+                        python-pip \
+                        ccontrol \
+                        virtualenv \
+                        libffi-dev \
+                        libssl-dev \
+                        libpython2.7-dev"])
+
+        # Technically you can create a virtualenv from within python
+        # but pip can only be run over tor on tails, and debugging that
+        # along with instaling a third-party dependency is not worth
+        # the effort here.
+        subprocess.call(['torify', 'virtualenv', VENV_DIR])
+
+    pip_install_cmd = ['torify', os.path.join(VENV_DIR, 'bin', 'pip'), 'install']
+    requirements_cmd = ['-r', os.path.join(ANSIBLE_PATH, 'requirements.txt'),
+                        '--require-hashes']
+
+    if args.update:
+        subprocess.call(pip_install_cmd+['-U']+requirements_cmd)
+    else:
+        subprocess.call(pip_install_cmd+requirements_cmd)
+
+def install_securedrop(args):
+    """Install/Update SecureDrop"""
+    activate_venv()
+    # Yaml library cannot be imported until virtualenv is activated
+    # (hence the yaml library is not imported up-top)
+    import yaml
+
+    try:
+        site_config_vars = yaml.safe_load(open(SITE_CONFIG, 'ro').read())
+
+        ansible_user = args.user if args.user else site_config_vars['ssh_users']
+
+        sudo_prompt = getpass.getpass(prompt="Sudo Pass [Blank if not needed]:")
+        subprocess.call(['./securedrop-prod.yml',
+                '-u '+ansible_user,
+                "-e ansible_become_pass={}".format(sudo_prompt if sudo_prompt else ""),
+                ],
+                cwd=ANSIBLE_PATH)
```

Comment:
A note on the password handling here: it's not quite clear to Admins whether the "if not needed" condition is satisfied. What we mean is that passwords are only required for the _first_ time the playbook is run (assuming no errors), since the playbook configures passwordless sudo for the admin user account.

We could detect whether a successful provisioning has already occurred by checking for e.g. `app-ssh-aths`, but that seems overwrought. 

Also, the `-e` option for passing in the sudo password could be an Ansible-specific env var. The typical preference for env vars or CLI options for sensitive values doesn't totally apply in Tails, but better safe than sorry.